### DISCUSSION TITLE
Is there a way to identify which cache is being initialized in the replacement policy code?

### DISCUSSION THREAD

**Discussion by Setu-Gupta** [ChampSim/ChampSim#426] [Q&A] [answered]
I'm trying to implement a replacement policy which reads some parameters from the environment. Depending on what cache (L1I, L1D, L2C or LLC) is being initialized, I want to read different environment variables. Since I would be using the same replacement policy for all the caches, I need to figure out for what cache is the policy being initialized. How can I find this out?

**Comment by GinoAC** [comment]
In the past I've done something simlar using the NAME string each cache is instantiated with (cache.h line 163ish). Someone else may have a better method but its an option.

**Reply by Setu-Gupta** [reply]
Yeah. That's one solution but I don't know if the naming scheme will remain the same even with later versions of ChampSim. That's why I was looking for something more stable.

**Reply by ngober** [reply, marked as answer]
Using the `NAME` member is a good idea. You can specify the name in the configuration as

    {
        "ooo_cpu": [ {
            "L1D": { "name": "my_l1d" }
        } ]
    }

That would be a stable way to do it. The caches are given default names if you don't supply them, but if you want to rely on them, you should go ahead and specify them.
